etapas/mapas/documentacion/man/man3/servidor_novo.3
.TH "servidor_novo" 3 "Martes, 7 de Agosto de 2012" "Version 0.1" "Generacion de Mapas" \" -*- nroff -*-
.ad l
.nh
.SH NAME
servidor_novo \- 
.PP
Servidor para el Pioneer P3-DX.  

.SH SYNOPSIS
.br
.PP
.SS "Funciones"

.in +1c
.ti -1c
.RI "def \fBrequestCallback\fP"
.br
.RI "\fISirve cuando se manda el comando 'test' en el paquete. \fP"
.ti -1c
.RI "def \fBmovimiento\fP"
.br
.RI "\fISirve cuando se manda el comando 'mover' en el paquete. \fP"
.ti -1c
.RI "def \fBrotar\fP"
.br
.RI "\fISirve cuando se manda el comando 'rotar' en el paquete. \fP"
.ti -1c
.RI "def \fBposicion\fP"
.br
.RI "\fISirve cuando se manda el comando 'pose' en el paquete. \fP"
.in -1c
.SS "Variables"

.in +1c
.ti -1c
.RI "tuple \fBrobot\fP = ArRobot()"
.br
.ti -1c
.RI "tuple \fBgyro\fP = ArAnalogGyro(\fBrobot\fP)"
.br
.ti -1c
.RI "tuple \fBsonarDev\fP = ArSonarDevice(7)"
.br
.ti -1c
.RI "tuple \fBserver\fP = ArServerBase()"
.br
.ti -1c
.RI "tuple \fBpacket\fP = ArNetPacket()"
.br
.ti -1c
.RI "tuple \fBcon\fP = ArSimpleConnector(sys.argv)"
.br
.ti -1c
.RI "tuple \fBserverInfoRobot\fP = ArServerInfoRobot(\fBserver\fP, \fBrobot\fP)"
.br
.ti -1c
.RI "tuple \fBserverInfoSensor\fP = ArServerInfoSensor(\fBserver\fP, \fBrobot\fP)"
.br
.ti -1c
.RI "tuple \fBdrawings\fP = ArServerInfoDrawings(\fBserver\fP)"
.br
.ti -1c
.RI "tuple \fBmodeStop\fP = ArServerModeStop(\fBserver\fP, \fBrobot\fP)"
.br
.ti -1c
.RI "tuple \fBmodeRatioDrive\fP = ArServerModeRatioDrive(\fBserver\fP, \fBrobot\fP)"
.br
.ti -1c
.RI "tuple \fBmodeWander\fP = ArServerModeWander(\fBserver\fP, \fBrobot\fP)"
.br
.ti -1c
.RI "tuple \fBcommands\fP = ArServerHandlerCommands(\fBserver\fP)"
.br
.ti -1c
.RI "tuple \fBuCCommands\fP = ArServerSimpleComUC(\fBcommands\fP, \fBrobot\fP)"
.br
.ti -1c
.RI "tuple \fBloggingCommands\fP = ArServerSimpleComMovementLogging(\fBcommands\fP, \fBrobot\fP)"
.br
.ti -1c
.RI "tuple \fBgyroCommands\fP = ArServerSimpleComGyro(\fBcommands\fP, \fBrobot\fP, \fBgyro\fP)"
.br
.ti -1c
.RI "tuple \fBconfigCommands\fP = ArServerSimpleComLogRobotConfig(\fBcommands\fP, \fBrobot\fP)"
.br
.in -1c
.SH "Descripción detallada"
.PP 
Servidor para el Pioneer P3-DX. 

Se lo puede utilizar con multiples conexiones de clientes trabaja en el puerto 7272 La lista de comandos para el paquete ArNetPacket se encuentra en el anexo 1 
.PP
\fBAutores:\fP
.RS 4
Danny Vasconez 
.PP
Daniel Granda 
.RE
.PP
\fBVersión:\fP
.RS 4
0.0.2 
.RE
.PP
\fBFecha:\fP
.RS 4
2012 
.RE
.PP
\fBPrecondición:\fP
.RS 4
Tener conectada la plataforma Pioneer P3-DX. 
.RE
.PP
\fBBug\fP
.RS 4
Nada 
.RE
.PP
\fBAtención:\fP
.RS 4
uso inapropiado puede hacer que la aplicacion falle 
.RE
.PP

.SH "Documentación de las funciones"
.PP 
.SS "def servidor_novo.movimiento (client, packet)"
.PP
Sirve cuando se manda el comando 'mover' en el paquete. \fBParámetros:\fP
.RS 4
\fIclient\fP El cliente que manda el paquete 
.br
\fIpacket\fP el paquete que recibe el servidor para el comando 
.RE
.PP
\fBDevuelve:\fP
.RS 4
Nada 
.RE
.PP

.PP
Definición en la línea 62 del archivo servidor_novo.py.
.PP
.nf
63                              :
64   robot.lock()
65   robot.comInt(8,5000) #move(5000) para atras move(-4999)
  robot.unlock()
.fi
.SS "def servidor_novo.posicion (client, packet)"
.PP
Sirve cuando se manda el comando 'pose' en el paquete. \fBParámetros:\fP
.RS 4
\fIclient\fP El cliente que manda el paquete 
.br
\fIpacket\fP el paquete que recibe el servidor para el comando 
.RE
.PP
\fBDevuelve:\fP
.RS 4
ArNetPacket con la informacion en coordenadas X,Y,T al cliente 
.RE
.PP

.PP
Definición en la línea 84 del archivo servidor_novo.py.
.PP
.nf
85                            :
86   robot.lock()
87   poses = sonarDev.getCurrentBufferAsVector()
88   packet=ArNetPacket()
89   packet.doubleToBuf(len(poses))
90   for p in poses:
91     print p
92     valor=str(p).strip('(').strip(')').strip('X:')
93     valor=valor.split(',')
94     valor_x=valor[0]
95     valor_y=valor[1].strip().strip('Y:')
96     packet.byte4ToBuf(int(float(valor_x)))
97     packet.byte4ToBuf(int(float(valor_y)))
98   packet.finalizePacket()
99   print packet.verifyCheckSum()
100   client.sendPacketTcp(packet)
101   robot.unlock()
102 
103 # This example demonstrates how to use ArNetworking in Python. 
104 
105 # Global library initialization, just like the C++ API:
106 Aria.init()
107 
# Create a robot object:
.fi
.SS "def servidor_novo.requestCallback (client, packet)"
.PP
Sirve cuando se manda el comando 'test' en el paquete. \fBParámetros:\fP
.RS 4
\fIclient\fP El cliente que manda el paquete 
.br
\fIpacket\fP el paquete que recibe el servidor para el comando 
.RE
.PP
\fBDevuelve:\fP
.RS 4
Nada 
.RE
.PP

.PP
Definición en la línea 50 del archivo servidor_novo.py.
.PP
.nf
51                                    :
52   replyPacket = ArNetPacket()
53   replyPacket.strToBuf(str(robot.getPose().x));
54   print 'requestCallback received a packet with command #%d. Sending a reply...\n' % (packet.getCommand())
  client.sendPacketTcp(replyPacket)
.fi
.SS "def servidor_novo.rotar (client, packet)"
.PP
Sirve cuando se manda el comando 'rotar' en el paquete. \fBParámetros:\fP
.RS 4
\fIclient\fP El cliente que manda el paquete 
.br
\fIpacket\fP el paquete que recibe el servidor para el comando 
.RE
.PP
\fBDevuelve:\fP
.RS 4
Nada 
.RE
.PP

.PP
Definición en la línea 73 del archivo servidor_novo.py.
.PP
.nf
74                         :
75   robot.lock()
76   robot.comInt(12,50) 
  robot.unlock()
.fi
.SH "Documentación de las variables"
.PP 
.SS "tuple \fBservidor_novo.commands\fP = ArServerHandlerCommands(\fBserver\fP)"
.PP
Definición en la línea 162 del archivo servidor_novo.py.
.SS "tuple \fBservidor_novo.con\fP = ArSimpleConnector(sys.argv)"
.PP
Definición en la línea 134 del archivo servidor_novo.py.
.SS "tuple \fBservidor_novo.configCommands\fP = ArServerSimpleComLogRobotConfig(\fBcommands\fP, \fBrobot\fP)"
.PP
Definición en la línea 170 del archivo servidor_novo.py.
.SS "tuple \fBservidor_novo.drawings\fP = ArServerInfoDrawings(\fBserver\fP)"
.PP
Definición en la línea 151 del archivo servidor_novo.py.
.SS "tuple \fBservidor_novo.gyro\fP = ArAnalogGyro(\fBrobot\fP)"
.PP
Definición en la línea 111 del archivo servidor_novo.py.
.SS "tuple \fBservidor_novo.gyroCommands\fP = ArServerSimpleComGyro(\fBcommands\fP, \fBrobot\fP, \fBgyro\fP)"
.PP
Definición en la línea 168 del archivo servidor_novo.py.
.SS "tuple \fBservidor_novo.loggingCommands\fP = ArServerSimpleComMovementLogging(\fBcommands\fP, \fBrobot\fP)"
.PP
Definición en la línea 166 del archivo servidor_novo.py.
.SS "tuple \fBservidor_novo.modeRatioDrive\fP = ArServerModeRatioDrive(\fBserver\fP, \fBrobot\fP)"
.PP
Definición en la línea 156 del archivo servidor_novo.py.
.SS "tuple \fBservidor_novo.modeStop\fP = ArServerModeStop(\fBserver\fP, \fBrobot\fP)"
.PP
Definición en la línea 155 del archivo servidor_novo.py.
.SS "tuple \fBservidor_novo.modeWander\fP = ArServerModeWander(\fBserver\fP, \fBrobot\fP)"
.PP
Definición en la línea 157 del archivo servidor_novo.py.
.SS "tuple \fBservidor_novo.packet\fP = ArNetPacket()"
.PP
Definición en la línea 121 del archivo servidor_novo.py.
.SS "tuple \fBservidor_novo.robot\fP = ArRobot()"
.PP
Definición en la línea 108 del archivo servidor_novo.py.
.SS "tuple \fBservidor_novo.server\fP = ArServerBase()"
.PP
Definición en la línea 119 del archivo servidor_novo.py.
.SS "tuple \fBservidor_novo.serverInfoRobot\fP = ArServerInfoRobot(\fBserver\fP, \fBrobot\fP)"
.PP
Definición en la línea 149 del archivo servidor_novo.py.
.SS "tuple \fBservidor_novo.serverInfoSensor\fP = ArServerInfoSensor(\fBserver\fP, \fBrobot\fP)"
.PP
Definición en la línea 150 del archivo servidor_novo.py.
.SS "tuple \fBservidor_novo.sonarDev\fP = ArSonarDevice(7)"
.PP
Definición en la línea 114 del archivo servidor_novo.py.
.SS "tuple \fBservidor_novo.uCCommands\fP = ArServerSimpleComUC(\fBcommands\fP, \fBrobot\fP)"
.PP
Definición en la línea 164 del archivo servidor_novo.py.
.SH "Autor"
.PP 
Generado automáticamente por Doxygen para Generacion de Mapas del código fuente.


CHANGELOG
Version 0.1.6b 2011-xx-xx
------------------------
2 modulos xml para los temas (themes/<tema>/*.xml) para saber cuales
  son las etiquetas y valores a usar 
  <<si consigo esto sube a 0.1.7a-RC1 >>
  <<si consigo esto sube a 0.1.7a pero consiguiendo punto 3 y 2>>
3 modulos xml para enlistar los temas la etiqueta tema contiene el 
  nombre del arvhico svg (ej reloj/protector.svg) y la etiqueta el 
  nombre del tema "wallv2" quedando de esta manera
  <theme nombre="wallv2">reloj/protector.svg</theme>  
  <<<si consigo esto sube a 0.1.7a-RC1 >>>
  <<<si consigo esto sube a 0.1.7a pero consiguiendo punto 2 y 3>>>
<4 conseguir cambiar el fondo de pantalla y su GUI
  <<<si consigo esto subo a 0.1.6a>>>
  <<<si consigo esto sube a 0.1.7a-RC1  si consegi punto 2>>>
  <<<si consigo esto sube a 0.1.7a-RC1  si consegi punto 3>>>
  <<<si consigo esto sube a 0.1.7b  si consegi punto 2 y 3>>>
<5 Consiguir widget-pasivos en la libreria Ergus 
<6 Consiguir widget-activos en la libreria Ergus 
<7 Consiguir widget-interactivos en la libreria Ergus
8 Construir la GUI del clic izq que es para configuracion general y todos los puntos anteriores
  <<Sube a la version 0.2a>> 

Version 0.1.6b 2011-xx-xx
- Icarus sube a la version 0.1b integrado con icarus_fondo
- Demonios wallv3 y poweroff utilizan los archivos "protector.xml" de cada uno para poder saber informacion de dimensiones y posiciones del widgets
- icarus_panel version 0.1_alfa. Sirve como panel para el escritorio por el momento solo tiene un windows_selector

Version 0.1.6a 2011-07-15
- Corregido bugs instalador y del GUI
- Autostart multiple en icarus-gui en el archivo "demonio.xml"
- Anadido widget en modo pasivo, activo y interactivo a Ergus
- Icarus_demonio_wall sube a la version 0.3
- icarus_pantallazo sube a la version 0.2
- icarus_demonio_poweroff version 0.1 
- icarus_fondo version 0.1_alfa. Sirve como wallpaper-setter basado en csetroot de circus
- Anadido archivo "pantallazo.xml" para saber que opciones tiene pantallazo
- Modificado iconstatus de icarus-gui a
   *Anadir fondo de pantalla (icarus_fondo v0.1-alfa)
   *Seleccionar demonio (integra a "activar/desactivar demonio")
   *Ponar captura de pantalla (icarus_pantallazo v0.2)
   *Acerca de
   *Salir

Version 0.1.5, 2011-06-23
- anadido iconstatus para el tray del sistema con las opciones:
   *Activar demonio
   *Desactivar demonio
   *Seleccionar demonio
   *Ponar captura de pantalla (icarus_pantallazo v0.1)
   *Acerca de
   *Salir
- Anadido archivo "demonio.xml" para saber que demonios se tiene para utilizar  
- Icaurs_demonio_wall sube a la version 0.2

Version 0.1, 2011-06-07
----------------------
- "First" release.


etapas/fuzzy/servidor_novo.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#       servidor_novo.py
#       
#       Copyright 2012 Danny Enrique Vasconez <dannyvasconeze@gmail.com>
#                      Daniel Engiberto Granda Guitierrez <degranda87@gmail.com>
#       
#       This program is free software; you can redistribute it and/or modify
#       it under the terms of the GNU General Public License as published by
#       the Free Software Foundation; either version 2 of the License, or
#       (at your option) any later version.
#       
#       This program is distributed in the hope that it will be useful,
#       but WITHOUT ANY WARRANTY; without even the implied warranty of
#       MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#       GNU General Public License for more details.
#       
#       You should have received a copy of the GNU General Public License
#       along with this program; if not, write to the Free Software
#       Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
#       MA 02110-1301, USA.


from AriaPy import *
from ArNetworkingPy import *
import sys
"""\package servidor_novo
   \brief Servidor para el Pioneer P3-DX
   \details Se lo puede utilizar con multiples conexiones de clientes trabaja en el puerto 7272
   La lista de comandos para el paquete ArNetPacket se encuentra en el anexo 1
   \authors    Danny Vasconez
   \authors   Daniel Granda
   \version   0.0.2
   \date      2012
   \pre       Tener conectada la plataforma Pioneer P3-DX.
   \bug       Nada
   \warning   uso inapropiado puede hacer que la aplicacion falle
   
"""

def requestCallback(client, packet):
  """
  \brief Sirve cuando se manda el comando "test" en el paquete
  \param client El cliente que manda el paquete
  \param packet el paquete que recibe el servidor para el comando 
  \return Nada
  """
  replyPacket = ArNetPacket()
  replyPacket.strToBuf(str(robot.getPose().x));
  print "requestCallback received a packet with command #%d. Sending a reply...\n" % (packet.getCommand())
  client.sendPacketTcp(replyPacket)
def movimiento(client,packet):
  """
  \brief Sirve cuando se manda el comando "mover" en el paquete
  \param client El cliente que manda el paquete
  \param packet el paquete que recibe el servidor para el comando 
  \return Nada
  """
  robot.lock()
  robot.comInt(8,5000) #move(5000) para atras move(-4999)
  robot.unlock()
def rotar(client,packet):
  """
  \brief Sirve cuando se manda el comando "rotar" en el paquete
  \param client El cliente que manda el paquete
  \param packet el paquete que recibe el servidor para el comando 
  \return Nada
  """
  robot.lock()
  robot.comInt(12,50) 
  robot.unlock()
def posicion(client,packet):
  """
  \brief Sirve cuando se manda el comando "pose" en el paquete
  \param client El cliente que manda el paquete
  \param packet el paquete que recibe el servidor para el comando 
  \return ArNetPacket con la informacion en coordenadas X,Y,T al cliente
  """
  robot.lock()
  poses = sonarDev.getCurrentBufferAsVector()
  packet=ArNetPacket()
  packet.doubleToBuf(len(poses))
  for p in poses:
    print p
    valor=str(p).strip('(').strip(')').strip('X:')
    valor=valor.split(",")
    valor_x=valor[0]
    valor_y=valor[1].strip().strip('Y:')
    packet.byte4ToBuf(int(float(valor_x)))
    packet.byte4ToBuf(int(float(valor_y)))
  packet.finalizePacket()
  print packet.verifyCheckSum()
  client.sendPacketTcp(packet)
  robot.unlock()

# This example demonstrates how to use ArNetworking in Python. 

# Global library initialization, just like the C++ API:
Aria.init()

# Create a robot object:
robot = ArRobot()

# make a gyro 
gyro = ArAnalogGyro(robot)

#sonar, must be added to the robot
sonarDev = ArSonarDevice(7)
#add the sonar to the robot
robot.addRangeDevice(sonarDev)

# make the core server object:
server = ArServerBase()
#
packet = ArNetPacket()
#
#anadir comandos
server.addData("test", "move(30)", requestCallback, "none", "none")
server.addData("rotar", "move(30)", rotar, "none", "none")
server.addData("mover", "move30", movimiento, "none", "none")
server.addData("pose", "valores del sensor en X,Y,T", posicion, "none", "primero el numero de sensores y luego sus valores X,Y,T")
#
# Create a "simple connector" object and connect to either the simulator
# or the robot. Unlike the C++ API which takes int and char* pointers, 
# the Python constructor just takes argv as a list.
print "Connecting..."

con = ArSimpleConnector(sys.argv)
if (not con.connectRobot(robot)):
    print "Could not connect to robot, exiting"
    Aria.exit(1)

#
server.resetTracking()

#
# Open the server on port 7272:
if (not server.open(7272)):
    print "Could not open server, exiting"
    Aria.exit(1)

# Create various services and attach them to the core server
serverInfoRobot = ArServerInfoRobot(server, robot)
serverInfoSensor = ArServerInfoSensor(server, robot)
drawings = ArServerInfoDrawings(server)
drawings.addRobotsRangeDevices(robot)

# ways of moving the robot
modeStop = ArServerModeStop(server, robot)
modeRatioDrive = ArServerModeRatioDrive(server, robot)
modeWander = ArServerModeWander(server, robot)
modeStop.addAsDefaultMode()
modeStop.activate()

# set up the simple commands ("custom commands" in MobileEyes)
commands = ArServerHandlerCommands(server)
# add the simple commands for the microcontroller
uCCommands = ArServerSimpleComUC(commands, robot)
# add the simple commands for logging
loggingCommands = ArServerSimpleComMovementLogging(commands, robot)
# add the simple commands for the gyro
gyroCommands = ArServerSimpleComGyro(commands, robot, gyro)
# Add the simple command for logging the robot config and orig robot config
configCommands = ArServerSimpleComLogRobotConfig(commands, robot)
    
# Run the robot and server threads in the background:
print "Running..."
robot.runAsync(1)

server.runAsync()

robot.enableMotors()

robot.waitForRunExit()

#
#while (server.getRunningWithLock()):
#  ArUtil.sleep(1000)
#  server.broadcastPacketTcp(packet, "updateNumbers")
#

